### ISSUE TITLE
CopyCommentsFromBaseCodeRefactoringProvider.ComputeRefactoringsAsync XML parse exception

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#5161]
1. Start MonoDevelop, open new C# console app
2. Place caret on class name

Exception bar is shown: XmlException there are multiple root elements

![image](https://user-images.githubusercontent.com/679326/41752289-7bc5477e-757b-11e8-9990-be1896c86e30.png)

```
 	System.Xml.dll!System.Xml.XmlTextReaderImpl.Throw Line 2501	C#
 	System.Xml.dll!System.Xml.XmlTextReaderImpl.Throw Line 2477	C#
 	System.Xml.dll!System.Xml.XmlTextReaderImpl.ParseDocumentContent Line 3644	C#
 	System.Xml.dll!System.Xml.XmlTextReaderImpl.Read Line 1166	C#
 	System.Xml.Linq.dll!System.Xml.Linq.XContainer.ReadContentFrom Line 3083	C#
 	System.Xml.Linq.dll!System.Xml.Linq.XContainer.ReadContentFrom Line 3193	C#
 	System.Xml.Linq.dll!System.Xml.Linq.XDocument.Load Line 5730	C#
 	System.Xml.Linq.dll!System.Xml.Linq.XDocument.Parse Line 5779	C#
 	System.Xml.Linq.dll!System.Xml.Linq.XDocument.Parse Line 5751	C#
>	RefactoringEssentials.dll!RefactoringEssentials.CSharp.CodeRefactorings.CopyCommentsFromBaseCodeRefactoringProvider.ComputeRefactoringsAsync Line 45	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync Line 142	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringsAsync.AnonymousMethod__0 Line 105	C#

```

XML snippet in question:
```
      <summary>Supports all classes in the .NET Framework class hierarchy and provides low-level services to derived classes. This is the ultimate base class of all classes in the .NET Framework; it is the root of the type hierarchy.To browse the .NET Framework source code for this type, see the Reference Source.</summary>
      <filterpriority>1</filterpriority>
```

> VS bug [#636929](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/636929)

**Comment by mrward** [comment]
I saw this exception bar. Clicking the StackTrace button did not display the stack trace for me for some reason. The IDE log had:

```
Roslyn wanted to display an exception to user
System.Xml.XmlException: There are multiple root elements. Line 5, position 14.
  at System.Xml.XmlTextReaderImpl.Throw (System.Exception e) [0x00027] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml/System/Xml/Core/XmlTextReaderImpl.cs:2503 
  at System.Xml.XmlTextReaderImpl.Throw (System.String res, System.String arg) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml/System/Xml/Core/XmlTextReaderImpl.cs:2478 
  at System.Xml.XmlTextReaderImpl.Throw (System.Int32 pos, System.String res) [0x0000c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml/System/Xml/Core/XmlTextReaderImpl.cs:2466 
  at System.Xml.XmlTextReaderImpl.ParseDocumentContent () [0x001e5] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml/System/Xml/Core/XmlTextReaderImpl.cs:3650 
  at System.Xml.XmlTextReaderImpl.Read () [0x0008c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml/System/Xml/Core/XmlTextReaderImpl.cs:1168 
  at System.Xml.Linq.XContainer.ReadContentFrom (System.Xml.XmlReader r) [0x00201] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml.Linq/System/Xml/Linq/XLinq.cs:3085 
  at System.Xml.Linq.XContainer.ReadContentFrom (System.Xml.XmlReader r, System.Xml.Linq.LoadOptions o) [0x00005] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml.Linq/System/Xml/Linq/XLinq.cs:3090 
  at System.Xml.Linq.XDocument.Load (System.Xml.XmlReader reader, System.Xml.Linq.LoadOptions options) [0x00080] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml.Linq/System/Xml/Linq/XLinq.cs:5747 
  at System.Xml.Linq.XDocument.Parse (System.String text, System.Xml.Linq.LoadOptions options) [0x00016] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml.Linq/System/Xml/Linq/XLinq.cs:5797 
  at System.Xml.Linq.XDocument.Parse (System.String text) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System.Xml.Linq/System/Xml/Linq/XLinq.cs:5769 
  at RefactoringEssentials.CSharp.CodeRefactorings.CopyCommentsFromBaseCodeRefactoringProvider+<ComputeRefactoringsAsync>d__0.MoveNext () [0x00288] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeRefactorings/Synced/CopyCommentsFromBaseCodeRefactoringProvider.cs:45 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:152 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:447 
  at Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService+<GetRefactoringFromProviderAsync>d__8.MoveNext () [0x000a0] in /_/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:143 
```

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.348 (master)

Pull Request #5164 merged by: Mike Krüger
Author: mono
Commit: 6f1cac5124555ff5e278e41ec6bd4c6a2e22a6d5 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5164 merged by: Mike Kr??ger
Author: mono
Commit: 6f1cac5124555ff5e278e41ec6bd4c6a2e22a6d5 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-17] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #5164 merged by: Mike Kr??ger  
 Author: mono  
 Commit: 6f1cac5124555ff5e278e41ec6bd4c6a2e22a6d5 (xamarin/monodevelop)
